components/tidb_query_executors/src/fast_hash_aggr_executor.rs

```diff
@@ -237,6 +248,29 @@ pub struct FastHashAggregationImpl {
     group_by_exp: RpnExpression,
     group_by_field_type: FieldType,
     states_offset_each_logical_row: Vec<usize>,
+    /// Number of bytes allocated by the executor.
+    n_bytes: usize,
+}
+
+impl Drop for FastHashAggregationImpl {
+    fn drop(&mut self) {
+        MEMTRACE_QUERY_EXECUTOR
+            .aggr_fast_hash
+            .sub(self.n_bytes as i64);
+    }
+}
+
+impl MemoryTrace for FastHashAggregationImpl {
+    #[inline]
+    fn alloc_trace(&mut self, len: usize) {
+        self.n_bytes += len;
+        MEMTRACE_QUERY_EXECUTOR.aggr_fast_hash.add(len as i64);
```

Comment:
a general question: how is the prometheus gauge implemented internally to avoid the performance issue. Does it have sophisticated thread-local vars or it's just global variable?
If it's the latter, did you try to the run performance comparison with and without this change?

Reply:
It's not a thread local variable. Not sure who has to do the tests. Do we have some documentation about the HW, Config, types of tests etc. to run?

Reply:
we can use benchbot to launch the test comparison. Manual: https://pingcap.feishu.cn/wiki/wikcn72geTuUtMaJ6VC79IjaAbb#
Tips: use translation function of Feishu Doc to translate it into pure English version:)

Atomic operation is OK for ops around a 100K or so, but will be a bottleneck if the ops can reach 1M/s. 
I guess we're probably fine here. There may be very tiny performance overhead. < 0.5%

Reply:
Submitted.